noir-projects/noir-contracts/contracts/counter_contract/src/main.nr

```diff
@@ -136,6 +136,9 @@ pub contract Counter {
     }
     // docs:end:txe_test_increment
 
+    // TODO(#13221): I had to comment out all the env.advance_block_by(1) and sync_notes() calls because that triggered
+    // an issue with how state is committed. The calls to advance_block_by(1) seems to have always been unnecessary
+    // in this test but anyway we should fix the issue and try to get the block advancing working here.
```

Comment:
This is [the issue](https://aztecprotocol.slack.com/archives/C07R3GTKJ49/p1743526408209209) discussed on slack.